examples/kubernetes/static_provisioning_with_subpath/README.md

```diff
@@ -0,0 +1,54 @@
+## Static Provisioning Example
+This example shows how to make a pre-created FSx for Lustre filesystem mounted inside container.
+It uses the optional subpath parameter to map a sub-folder of the FSx for Lustre filesystem into the cluster, allowing one filesystem to be used independently by many volumes.
+
+### Edit [Persistent Volume Spec](./specs/pv.yaml)
+```
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: fsx-pv
+spec:
+  capacity:
+    storage: 1200Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteMany
+  mountOptions:
+    - flock
+  persistentVolumeReclaimPolicy: Recycle
+  csi:
+    driver: fsx.csi.aws.com
+    volumeHandle: [FileSystemId]
+    volumeAttributes:
+      dnsname: [DNSName]
+      mountname: [MountName]
+      subpath: someDirName
+```
+Replace `volumeHandle` with `FileSystemId`, `dnsname` with `DNSName` and `mountname` with `MountName`. You can get both `FileSystemId`, `DNSName` and `MountName` using AWS CLI:
+When the optional subpath parameter is set, this will create a folder with the name from subpath in the root directory of the FSx filesystem.
```

Comment:
I am a bit wary about this approach, it blurs the line between static/dynamic provisioning. Strictly speaking I would expect static to mean every property of this volume already exists, including the subpath in it.

What if I set a `subPath` in my pod's volumeMount? If that subPath doesn't exist, the mount fails right?

So the difference is that if i set subpath in volumeAttributes the mount will be guaranteed to work because the driver will create it?

Reply:
Yeah, it exists (mostly) in support of dynamic provisioning many pv on one fsx.
Maybe there is some way to avoid it, but it seems that the subPath is on the mount so if I want to dynamically provision separate volumes on the same filesystem, I need a way to curry the intent to the node diver.

Reply:
I got it, I agree this is useful now I am just mulling over the api/syntax.

Your version of `subpath` has two implications, 1 is that the subpath will be created, 2 is that the subpath will be mounted (same as if I put subPath in my volumeMount), I think it would be clearer if these 2 things were split out with clearer intent. 

To solve 2, the EFS driver puts the path in the volumeHandle like fs-id:/path. The issue with using a volumeAttribute to store the path we discovered with EFS is it can lead to Pod launch failure if a Pod tries to consume 2 volumes with the same `volumeHandle` but different attributes. A volumeHandle alone must uniquely identify a volume, so if Kubernetes sees 

volume[0]
volumeHandle: fs-a
volumeAttribute: subpath: a
volume[1]
volumeHandle: fs-a
volumeAttribute: subpath: b

It won't bother to mount volume[1] even though it logically represents a different volume.

But thjis works as users expect:
volume[0]
volumeHandle: fs-a:/a
volume[1]
volumeHandle: fs-a:/b

To solve 1, I would like a volumeAttribute similar to HostPath's Directory or DirectoryOrCreate https://kubernetes.io/docs/concepts/storage/volumes/, and I would like the solution to be the same for all our CSI drivers to be consistent